### ISSUE TITLE
Review async tests to return promises instead of using a done callback

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#45209]
**tl;dr:**
* we can avoid the `done` function in tests by returning a `promise` instead
* this ensures that errors are properly handled by mocha when occurring
* it is important to return each promise (including nested ones) properly, otherwise you get unexpected promise rejection errors that are not resulting in a test failure!

**For example:**

```ts
	test('renameFolder', function (done: () => void) {
		let event: FileOperationEvent;
		const toDispose = service.onAfterOperation(e => {
			event = e;
		});

		const resource = uri.file(path.join(testDir, 'deep'));
		service.resolveFile(resource).done(source => {
			return service.rename(source.resource, 'deeper').then(renamed => {
				assert.equal(fs.existsSync(renamed.resource.fsPath), true);
				assert.equal(fs.existsSync(source.resource.fsPath), false);

				assert.ok(event);
				assert.equal(event.resource.fsPath, resource.fsPath);
				assert.equal(event.operation, FileOperation.MOVE);
				assert.equal(event.target.resource.fsPath, renamed.resource.fsPath);
				toDispose.dispose();

				done();
			});
		});
	});
```

Notice how if there is an error in `service.resolveFile()`, `done` won't ever be called.

The right way to run these is to simply return the promise and to remove the `done` argument from the function.

Making a pass over test that needs to be reviewed:
* [x] `treeModel.test.ts` @joaomoreno 
* [ ] `search.test.ts` @roblourens @chrmarti 
* [x] `textSearch.integrationTest.ts` @roblourens @chrmarti

**Comment by pradeepmurugesan** [comment]
@joaomoreno I know this is simple :-) .. Are you guys open for a PR ?

**Comment by bpasero** [comment]
@joaomoreno fixed most, left some, I updated the issue

**Comment by roblourens** [comment]
The search code is still callbacks all the way down, so I'm leaving those tests for now. There's actually an open PR that rewrote the search code to use Promise, but it was abandoned by the author with some merge conflicts. I should just finish it myself.
